internal/resultstore/resultstore.go

```diff
@@ -39,56 +39,73 @@ func ConstructPath() Option {
 	return option(func(rs *ResultStore) { rs.constructPath = true })
 }
 
-// BasePath sets the base path used while saving files to storage.
-func BasePath(base string) Option {
-	return option(func(rs *ResultStore) { rs.basePath = base })
-}
-
+// New creates a new ResultStore instance with the given bucket URL and options.
+// If the bucket URL is invalid, a nil pointer is returned.
 func New(bucket string, options ...Option) *ResultStore {
+	bucketURL, err := url.Parse(bucket)
+	if err != nil {
+		return nil
+	}
+
+	if bucketURL.Scheme == "file" {
+		// https://github.com/google/go-cloud/issues/3294
+		params := bucketURL.Query()
+		params.Set("no_tmp_dir", "true")
+		bucketURL.RawQuery = params.Encode()
+	}
+
 	rs := &ResultStore{
-		bucket: bucket,
+		bucket: bucketURL,
 	}
+
 	for _, o := range options {
 		o.set(rs)
 	}
 	return rs
 }
 
 func (rs *ResultStore) String() string {
-	s := rs.bucket + "/" + rs.basePath
+	// label when bucket path is constructed from package name
 	if rs.constructPath {
-		s += "+"
+		return rs.bucket.JoinPath("<dynamic path>").String()
 	}
-	return s
+
+	return rs.bucket.String()
 }
 
-func (rs *ResultStore) generatePath(p Pkg) string {
-	uploadPath := rs.basePath
-	if rs.constructPath {
-		uploadPath = path.Join(uploadPath, p.EcosystemName(), p.Name())
+// generateKey creates an identifier key to store an object with.
+// If p is non-nil and the ResultStore was constructed with
+// the ConstructPath() option, then the base key will be prefixed
+// with the ecosystem and name of the given package (in that order).
+// Otherwise, the basename is returned.
+func (rs *ResultStore) generateKey(baseKey string, p Pkg) string {
```

Comment:
I suggest reordering the arguments to better reflect the order they are joined together.

Reply:
done